### ISSUE TITLE
OSX issues linking to libraries

### ISSUE THREAD

**Issue by virtuald** [robotpy/robotpy-build#2]
Loading libraries dynamically via ctypes.LoadLibrary doesn't resolve dependencies, even if you load the dependent libraries first. Seems to be an OSX specific thing, and other python libraries have ran into this ([tensorflow](https://github.com/tensorflow/tensorflow/issues/6729))

Indeed, even the default pybind11 library that was generated by robotpy-build isn't able to find the library that it linked to. I suspect this could be addressed by changing the import path.

One potential solution is to install libraries to ~/lib: https://github.com/tensorflow/tensorflow/issues/6729#issuecomment-283545793 ... this is doable with a post-install step but we would want to properly version the library file if we took that approach.

Other potential solutions include changing the libraries via `install_name_tool`.

**Comment by virtuald** [comment]
I believe what we should do is use [macholib](https://pypi.org/project/macholib/) to edit the downloaded binaries instead of using `install_name_tool`. I believe we can set multiple rpath entries on OSX, so we could set it to work in a virtualenv and development environment by default, and hopefully that would be good enough.

Not sure if I'll have time to look at this. A workaround should be to set DYLD_LIBRARY_PATH, but there are caveats to that also.

**Comment by virtuald** [comment]
This might be an answer also: https://blog.tim-smith.us/2015/09/python-extension-modules-os-x/

**Comment by virtuald** [comment]
I found today that if you create symlinks to the binaries in ~/lib, then the dynamic loader can find the library. Unfortunately, there's no version information associated with the libraries, and too much potential for conflict.

I tested modifying the binaries via install_name_tool, and that fixes the problem. However, that tool is only installed on developer machines -- but I believe macholib could be used to do the same thing. I believe setting locations using a relative `@loader_path` makes the most sense. Would need to set the path differently based on whether we were installing or building a wheel -- and probably need a dev tool to find all the libs and 'fix them'.
